dbms/src/Storages/KVStore/Region.cpp

```diff
@@ -387,4 +401,48 @@ Timestamp Region::getLastObservedReadTso() const
     return last_observed_read_tso.load();
 }
 
+void Region::setRegionTableCtx(RegionTableCtxPtr ctx) const
+{
+    data.setRegionTableCtx(ctx);
+}
+
+void Region::maybeWarnMemoryLimitByTable(TMTContext & tmt, const char * from)
+{
+    // If there are data flow in, we will check if the memory is exhaused.
+    auto limit = tmt.getKVStore()->getKVStoreMemoryLimit();
+    auto current = real_rss.load();
```

Comment:
* If KVStoreMemoryLimit (max_memory_usage_for_all_queries) is explicitly set to 0, now it  will not print any logging. IMO, for this case we should use the server memory capacity * 80% as the warning limit.
* Only print log when the region_table_size > 50% * current is too large. I think we may easily miss the logging for  track. I recommend to lower the table level memory threshold to min(20% * current, 10GiB) to increase the recall rate for logging down info.

Reply:
- OK
- We'd better substract "start_time" memory.